### PR TITLE
Strip Pundit usage from admin controllers

### PR DESCRIPTION
**Pull request by vkrmis**
**Story card:** [ch1447](https://app.clubhouse.io/simpledotorg/story/1447/remove-pundit-usage-in-the-admin-controllers)

Removes Pundit permissions from the following controllers and their views

## This addresses

- [x] Admin::FacilitiesController
- [x] Admin::FacilityGroupsController
- [x] Admin::OrganizationsController
- [x] Admin::ProtocolsController
- [x] Admin::ProtocolDrugsController
- [x] Admin::UsersController
- [x] AdminsController

### CODE REVIEW

File: app/models/user_access.rb [hunk 1]
```diff
@@ -93,7 +93,7 @@ def accessible_admins(action)
   end
 
   def accessible_users(action)
-    return User.none unless [:manage, :view_reports].include?(action)
+    return User.none unless [:manage, :view_reports, :view_pii].include?(action)
```
Comment (harimohanraj89) [inline app/models/user_access.rb:96]: Nice. Is this an unrelated fix?

File: app/views/admin/facilities/index.html.erb [hunk 2]
```diff
@@ -96,15 +64,7 @@
...
-                <% if current_admin.permissions_v2_enabled? %>
-                  <%= link_to facility.name, [:admin, facility_group, facility], class: "d-block" %>
-                <% else %>
-                  <% if policy([:manage, :facility, facility]).show? %>
-                    <%= link_to facility.name, [:admin, facility_group, facility], class: "d-block" %>
-                  <% else %>
-                    <%= facility.name %>
-                  <% end %>
-                <% end %>
+                <%= link_to facility.name, [:admin, facility_group, facility], class: "d-block" %>
```
Comment (harimohanraj89) [inline app/views/admin/facilities/index.html.erb:67]: Might be a bug from before - do we not need to show the unhyperlinked names? Does our new scoping ensure we only loop through facilities you have access to, basically?

File: app/views/admin/users/index.html.erb [hunk 3]
```diff
@@ -46,16 +36,7 @@
...
-                <% if current_admin.permissions_v2_enabled? %>
-                    <%= link_to facility.name, [:admin, facility.facility_group, facility], class: "text-dark" %>
-                <% else %>
-                  <% if policy([:manage, :facility, facility]).show? %>
-                    <%= link_to facility.name, [:admin, facility.facility_group, facility], class: "text-dark" %>
-                  <% else %>
-                    <%= user.registration_facility.name %>
-                  <% end %>
-                <% end %>
+                <%= link_to facility.name, [:admin, facility.facility_group, facility], class: "text-dark" %>
```
Comment (harimohanraj89) [inline app/views/admin/users/index.html.erb:39]: Same question as before - does our new scoping ensure you'll have access to this facility?

File: app/views/admin/users/show.html.erb [hunk 4]
```diff
@@ -1,58 +1,54 @@
...
+  <div class="card-footer">
+    First login: <%= @user.logged_in_at.present? ? l(@user.logged_in_at) : '&lt;Not logged in yet&gt;'.html_safe %>
+  </div>
 </div>
 
-<% if policy([:cohort_report, @user.registration_facility]).view_health_worker_activity? %>
-<%= render 'shared/recent_bp_log',
-           blood_pressures: @recent_blood_pressures,
-           display_model: :user %>
+<% if current_admin.accessible_users(:view_pii).include?(@user) %>
```
Comment (harimohanraj89) [inline app/views/admin/users/show.html.erb:50]: Haha guess we missed this until now 😅

File: app/views/admins/_form.html.erb [hunk 5]
```diff
@@ -1,109 +1,89 @@
...
   </div>
+</div>
 
-  <script>
-    new AdminAccessEdit("facility-access").initialize()
-  </script>
-<% else %>
-  <%= javascript_include_tag "standalone/react_components" %>
-  <% user_permissions = current_admin.user_permissions.pluck(:permission_slug) %>
-  <%= react_component "InviteAdminForm",
```
Comment (harimohanraj89) [inline app/views/admins/_form.html.erb:94]: Ooh, can we delete some react files now too?

**Review by harimohanraj89 — APPROVED** [review]
My favorite part of this PR is how easy the test updates were - mostly just test setup data tweaks. Nice!

Some non-blocking thoughts and comments. Looks slick! ✨

File: app/models/user_access.rb (hunk 1 above)
Comment (vkrmis) [reply app/models/user_access.rb:96]: Not a 'fix' per se, this adds the `view_pii` action for `accessible_users`. Needed it for the `users#show` view.

File: app/views/admin/users/index.html.erb (hunk 3 above)
Comment (vkrmis) [reply app/views/admin/users/index.html.erb:39]: Yes

File: app/views/admins/_form.html.erb (hunk 5 above)
Comment (vkrmis) [reply app/views/admins/_form.html.erb:94]: 👻

File: app/views/admin/facilities/index.html.erb (hunk 2 above)
Comment (vkrmis) [reply app/views/admin/facilities/index.html.erb:67]: ✅
